src/renderers/shared/fiber/ReactFiberContext.js

```diff
@@ -0,0 +1,108 @@
+/**
+ * Copyright 2013-present, Facebook, Inc.
+ * All rights reserved.
+ *
+ * This source code is licensed under the BSD-style license found in the
+ * LICENSE file in the root directory of this source tree. An additional grant
+ * of patent rights can be found in the PATENTS file in the same directory.
+ *
+ * @providesModule ReactFiberContext
+ * @flow
+ */
+
+'use strict';
+
+import type { Fiber } from 'ReactFiber';
+
+var emptyObject = require('emptyObject');
+var invariant = require('invariant');
+var {
+  getComponentName,
+} = require('ReactFiberTreeReflection');
+var {
+  ClassComponent,
+} = require('ReactTypeOfWork');
+
+if (__DEV__) {
+  var checkReactTypeSpec = require('checkReactTypeSpec');
+}
+
+let index = -1;
+const contextStack : Array<Object> = [];
+const didPerformWorkStack : Array<boolean> = [];
```

Comment:
In this case it'll (probably) be more efficient to use an array since we won't need to reorder anything and we won't need to reallocate anything when it grows and shrinks.